src/script/sign.cpp

```diff
@@ -104,6 +152,85 @@ static bool CreateSig(const BaseSignatureCreator& creator, SignatureData& sigdat
     return false;
 }
 
+static bool CreateTaprootScriptSig(const BaseSignatureCreator& creator, SignatureData& sigdata, const SigningProvider& provider, std::vector<unsigned char>& sig_out, const XOnlyPubKey& pubkey, const uint256& leaf_hash, SigVersion sigversion)
+{
+    auto lookup_key = std::make_pair(pubkey, leaf_hash);
+    auto it = sigdata.taproot_script_sigs.find(lookup_key);
+    if (it != sigdata.taproot_script_sigs.end()) {
+        sig_out = it->second;
+    }
```

Comment:
In ea07a01b0cf7256304b17508debf74d94cccdac7 "Basic Taproot signing logic in script/sign.cpp"

Can you explain what this is doing and why?

Reply:
It's the equivalent of `SignatureData::signatures`, but for Taproot/Schnorr signatures. They're not indexed by CKeyId, but by full (untweaked) pubkey and leaf they occur in.

This is redundant for now, as there is no multiparty signing whatsoever, and no way of communicating partial signatures (needs PSBT extension), but this is pretty much what it would look like. I could drop it for this PR.